app/controllers/demo_mode/sessions_controller.rb

```diff
@@ -26,7 +26,7 @@ def new
     def create
       @session = Session.new(create_params)
       @session.save_and_generate_account_later!(**options_params.to_unsafe_h.deep_symbolize_keys)
-      @session.reload if ActiveJob::Base.queue_adapter_name == 'test'
+      @session.reload if ActiveJob::Base.queue_adapter_name == 'inline'
```

Comment:
Why do we need to reload here? If we don't reload, won't DemoMode poll the `#show` action and eventually get the reloaded attributes?

It seems odd that the controller would inspect our job queue configuration to determine if tests are currently running?

Reply:
There are use cases where we want the attributes back inline instead of relying on the polling from the show action. Also, setting the adapter to `:inline` in the consumer config does not override the behavior to execute the job inline. cc: @smudge

Reply:
At the very least, should we be checking `DemoMode::AccountGenerationJob.queue_adapter_name`? There's no guarantee that it's using the same queue adapter as `ActiveJob::Base`.

Reply:
condition removed, and it does seem much cleaner now.